### ISSUE TITLE
Investigates Flutter voice control support for iOS

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#75887]
We need to make sure flutter is accessible when uses iOS voice control

action items:

- [x] flutter does not enable semantics when voice control is turned on. https://github.com/flutter/engine/pull/24640
- [x] Delete line in text field does not delete the whole line. https://github.com/flutter/flutter/issues/77274
- [ ] Redo/Undo does not work. https://github.com/flutter/flutter/issues/77614
- [x] scrolling does not work. https://github.com/flutter/flutter/issues/80711

**Comment by pedromassangocode** [comment]
[triage-team]
(I'm assuming you mean to add iOS label)

**Comment by chunhtai** [comment]
bump the priority because this is something we needed to get done by the end of this quarter, otherwise it block our customers

**Comment by chunhtai** [comment]
It looks like flutter does enable semantics when voice control is on, we may need to hook it up in the engine

**Comment by chunhtai** [comment]
Some more update:

The good
It seems like we just need to make sure we provide accessibility label and everything should just work

The bad
I can't find a way to get whether the voice control is on. There doesn't seem to be a public API to query this data.
I tried to inspect what notification it sends when voice control is toggled on or off. It sends multiple notification with name
```
com.apple.accessibility.application.status
```
with empty userInfo, so we can't tell what status has changed.

I will keep investigating.

**Comment by chunhtai** [comment]
One workaround is to add a callback in [FlutterView isAccessibilityElement]. If this method is called, it is likely one of the assistive technologies is turned on, and we can just enable the semantics in the callback. This however cannot detect when the voice control is turned off, the semantics will continue to be collected.

I think this is the best we can do until iOS provides an API to listen for voice control is turned on or off.

**Comment by chunhtai** [comment]
I am going to close this issue and use individual issues to track the progress
